transfer/src/main/java/com/kin/ecosystem/transfer/AccountInfoManager.java

```diff
@@ -0,0 +1,86 @@
+package com.kin.ecosystem.transfer;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.v4.content.FileProvider;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+
+class AccountInfoManager {
+
+    interface InitCompletionListener {
+        void onSuccess();
+
+        void onError(String reason);
+    }
+
+    private static String FILE_NAME = "accountInfo.txt";
+    private static String FILE_PROVIDER_NAME = "KinTransferAccountInfoFileProvider";
+    private static String FILE_PROVIDER_DIR_NAME = "kintransfer_account_info";
+
+    private File file;
+
+    //need to be called first
+    void init(@NonNull Context context, @NonNull String publicAddress, @NonNull InitCompletionListener listener) {
+        if(publicAddress.isEmpty()){
+            listener.onError("public address cant be empty");
+            return;
+        }
+        String dirPath = context.getFilesDir().getAbsolutePath() + File.separator + FILE_PROVIDER_DIR_NAME;
+        File dir = new File(dirPath);
+        if (!dir.exists()) {
+            if (!dir.mkdir()) {
+                listener.onError("Cant create directory " + dirPath);
+                return;
+            }
+        }
+        String fileFullPath = dirPath + File.separator + FILE_NAME;
+        try {
+            file = new File(fileFullPath);
+            if (!file.exists()) {
+                if (!file.createNewFile()) {
+                    file = null;
+                    listener.onError("Cant create file " + fileFullPath);
+                    return;
+                }
+            }
+            FileWriter writer = new FileWriter(file);
+            writer.append(publicAddress);
+            writer.flush();
+            writer.close();
+        } catch (IOException e) {
+            file = null;
+            listener.onError("Cant write to file " + e.getMessage());
+            return;
+        }
+        listener.onSuccess();
+    }
+
+    //Can be called anytime (before init complete), when user decline transfer
+    void respondCancel(@NonNull Activity activity) {
+        Intent intent = new Intent();
+        activity.setResult(Activity.RESULT_CANCELED, intent);
+    }
+
+    //Can be called only after init returned success, hence file is not null
+    void respondOk(@NonNull Activity activity) {
+        if (file != null) {
+            Intent intent = new Intent();
+            String authority = activity.getPackageName() + "." + FILE_PROVIDER_NAME;
+            final Uri uri = FileProvider.getUriForFile(activity, authority, file);
+            intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
+                activity.revokeUriPermission(uri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
```

Comment:
why is this needed ?

Reply:
the way file provider ask for grand read permission

Reply:
doesn't make sense - revoking permission?